File: sdk-python/examples/properties/auto_paginate.py
"""Run with: python examples/properties/auto_paginate.py

Walk every contact property for the host with the auto-paginator. Pages are
fetched lazily - one HTTP call per page, only when the previous page's
buffer drains.
"""

from __future__ import annotations

from threecommon import ThreeCommon
from threecommon.properties import ListParams


def main() -> None:
    total = 0
    last_name = ""
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        for prop in client.properties.list_auto_paginate(ListParams(object_type="contact")):
            total += 1
            last_name = prop.name
            if total % 100 == 0:
                print(f"...processed {total} properties")

    print(f"walked {total} contact properties total (last: {last_name})")


if __name__ == "__main__":
    main()

File: sdk-node/examples/properties/auto-paginate.ts
/**
 * Iterate every contact property, transparently fetching each page as the
 * previous one drains.
 *
 * Run:
 *   npx tsx examples/properties/auto-paginate.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

let count = 0
let archived = 0
for await (const property of client.properties.listAutoPaginate({ objectType: 'contact' })) {
  count += 1
  if (property.status === 'archived') archived += 1
}

console.log(`iterated ${String(count)} contact properties`)
console.log(`${String(archived)} are archived`)
